src/persist-client/src/read.rs

```diff
@@ -720,6 +720,15 @@ where
     ) -> Result<Vec<LeasedBatchPart<T>>, Since<T>> {
         let batches = self.machine.snapshot(&as_of).await?;
 
+        if !PartialOrder::less_equal(self.since(), &as_of) {
+            error!(
+                "ReadHandle::snapshot called with as_of {:?} not past the since {:?}, \
```

Comment:
We actually saw this pop up for the persist-txn stuff in sentry. I wonder if it's doing something wrong? https://github.com/MaterializeInc/materialize/issues/24692

Reply:
Ugh, that was actually an older version of the code where I forgot a `!`. (It's totally fine if the since and as-of are both `[0]`.) A bit weird that we're getting issues from sentries from draft PRs in CI!

Reply:
> Ugh, that was actually an older version of the code where I forgot a !. (It's totally fine if the since and as-of are both [0].)

Oh good!

> A bit weird that we're getting issues from sentries from draft PRs in CI!

Dennis dug into that and [figured out what was going on](https://materializeinc.slack.com/archives/CTESPM7FU/p1706198776564169?thread_ts=1706181495.486039&cid=CTESPM7FU). Fixed going forward